### ISSUE TITLE
Quote any value if needed

### ISSUE THREAD

**Issue by tg** [sirupsen/logrus#301]
Currently (per [this switch](https://github.com/Sirupsen/logrus/blob/master/text_formatter.go#L142)) only `string` and `error` types in `TextFormatter` are subject to quoting. As logrus allows for logging string representations of _any_ type, instead of privileging just two "most common" types, quoting should be applied for every value which needs it. 

This is indeed introducing quite a lot of mess:

``` go
logrus.WithField("key", time.Now()).Print("stupid spaces")
// Output:
// time="2015-11-27T17:28:28+01:00" level=info msg="stupid spaces" key=2015-11-27 17:28:28.766592205 +0100 CE

logrus.WithField("key", struct{ string }{"v x=y"}).Print("oops, not a field")
// Output:
// time="2015-11-27T17:29:28+01:00" level=info msg="oops, not a field" key={v x=y}
```

**Comment by aybabtme** [comment]
hey @tg, I think you're right. If you can submit a PR with a fix, I'll be happy to review it.

**Comment by dmathieu** [comment]
This was fixed in #583.
